pycuda/scan.py

```diff
@@ -381,7 +381,7 @@ def __init__(
         self.scan_intervals_knl = scan_intervals_prg.get_function(
             name_prefix + "_scan_intervals"
         )
-        self.scan_intervals_knl.prepare("PIIPP")
+        self.scan_intervals_knl.prepare("PnIPP")
```

Comment:
I think you want `N` here, not `n`.

Reply:
Thanks, I initially took `n` as the index is supposed to be unsigned, but with 64 bits there should be no problems switching to int64 rather than uint64.

Reply:
Do I have this backwards? `pydoc struct` tells me
```
      n:ssize_t; N:size_t;
```
But your committee suggests this went the opposite way:
> [Switch from size_t to ssize_t](https://github.com/inducer/pycuda/pull/383/commits/607026fa7ca387b0f0c786587feb7a1679212cfc)

My thought was that `size_t` was what we want. Do you agree?

Reply:
You are right! I was the one to have it backward.

In `src/wrapper/_pvt_struct_v3.cp`, we have:

 ```C++
        {'n',       sizeof(size_t), SIZE_T_ALIGN,   nu_ssize_t,     np_ssize_t},
        {'N',       sizeof(size_t), SIZE_T_ALIGN,   nu_size_t,      np_size_t},
```
and `numpy.ctypeslib.ctypes.c_size_t` is `ctypes.c_ulong` while `numpy.ctypeslib.ctypes.c_ssize_t` is `ctypes.c_long`

Sorry for the noise!